src/main/java/de/progen_bot/commands/user/Say.java

```diff
@@ -1,32 +1,38 @@
 package de.progen_bot.commands.user;
 
+import java.time.Instant;
+
 import de.progen_bot.command.CommandHandler;
 import de.progen_bot.command.CommandManager.ParsedCommandString;
 import de.progen_bot.db.entities.config.GuildConfiguration;
+import de.progen_bot.listeners.SayListener;
+import de.progen_bot.listeners.SayModel;
 import de.progen_bot.permissions.AccessLevel;
+import net.dv8tion.jda.api.EmbedBuilder;
 import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
 
-
 public class Say extends CommandHandler {
 
     public Say() {
         super("say", "say <text>", "let the bot write a message");
     }
 
     @Override
-    public void execute(ParsedCommandString parsedCommand, MessageReceivedEvent event, GuildConfiguration configuration) {
-        event.getMessage().delete();
-        StringBuilder out = new StringBuilder(" ");
-        for (String s : parsedCommand.getArgs()) {
-            out.append(s).append(" ");
-        }
-
-        event.getTextChannel().sendMessage(out.toString()).queue();
+    public void execute(ParsedCommandString parsedCommand, MessageReceivedEvent event,
+            GuildConfiguration configuration) {
+
+        final String out = String.join(" ", parsedCommand.getArgs());
+
+        final SayModel sayModel = new SayModel(event.getTextChannel().getId(), new EmbedBuilder().setDescription(out)
+                .setAuthor(event.getAuthor().getAsTag()).setTimestamp(Instant.now()));
+        SayListener.addSayModel(event.getAuthor().getId(), sayModel);
+
+        event.getAuthor().openPrivateChannel()
+                .queue(channel -> channel.sendMessage("Please choose a color in hex <#fff> or <#ffffff>").queue());
```

Comment:
Sending DMs to users, which have DM disabled would fail at this point. Error handling using the failure callback/consumer would be necessary to avoid further exceptions during the creation of the embed.